convert2rhel/unit_tests/cert_test.py

```diff
@@ -98,30 +98,35 @@ def test_install_cert(self):
         shutil.rmtree(unit_tests.TMP_DIR)
 
 
+def test_remove_cert(caplog, remove_cert_setup):
+    cert_file = system_cert._target_cert_path
+    with open(cert_filename, 'wb') as  cert_file:
+        cert_file.write(b'some content')
+
+    system_cert[sys_cert].remove()
+
+    assert "/filename removed" in caplog.messages[-1]
+
+
 @pytest.mark.parametrize(
     (
-        "create_file",
+        "error_condition",
         "expected_text_in_logs",
     ),
     (
-        (True, "/filename removed"),
-        (False, "No such file or directory"),
+        (
+            OSError(2, "No such file or directory"),
+            "No such file or directory",
+        ),
+        (OSError(13, "[Errno 13] Permission denied: '/tmpdir/certfile'"), "Permission denied:"),
     ),
 )
-def test_remove_cert(
-    monkeypatch,
-    create_file,
-    expected_text_in_logs,
-    caplog,
-    tmpdir,
-):
-    tmp_file = tmpdir / "filename"
-    if create_file:
-        tmp_file.write(b"some content")
-
-    monkeypatch.setattr(cert.SystemCert, "_get_cert", value=mock.Mock(return_value=("anything", "anything")))
-    monkeypatch.setattr(cert.SystemCert, "_get_target_cert_path", value=mock.Mock(return_value=str(tmp_file)))
-
-    sys_cert = cert.SystemCert()
-    sys_cert.remove()
-    assert expected_text_in_logs in caplog.messages[-1]
+def test_remove_cert_error_conditions(error_condition, expected_text_in_logs, caplog, monkeypatch, remove_cert_setup):
+    def fake_os_remove(path):
+        raise error_condition
+
+    monkeypatch.setattr(os, "remove", fake_os_remove)
+
+    remove_cert_setup["sys_cert_instance"].remove()
```

Comment:
Where does `sys_cert_instance` come from in this case?

Reply:
sys_cert_instance is also leftover code from the old function remove_cert_setup and is apart of the dicionary that is returned from that old function. I have already fixed this in https://github.com/oamg/convert2rhel/pull/447/commits/204a0b5aa81e9b95bdebeb71da918e2559cf283f this commit!